### ISSUE TITLE
Conversion error for nullable int field

### ISSUE THREAD

**Issue by SpeedSX** [prisma/tiberius#78]
I have a table like this:
```sql
CREATE TABLE Test (
	id int IDENTITY(1,1) NOT NULL,
	value int NULL
)
```
Filled it with values:
```sql
INSERT INTO Test (value) VALUES (1)
INSERT INTO Test (value) VALUES (NULL)
```
and trying to read:
```rust
    let result = con.simple_query("select * from Test").await.unwrap();
    let rows: Vec<Row> = result.into_first_result().await.unwrap();
    info!("{:?}", rows);
    let a: Option<i32> = rows[0].try_get("value").unwrap();
    info!("a = {:?}", a);
    let b: Option<i32> = rows[1].try_get("value").unwrap();
    info!("b = {:?}", b);
```
Surprisingly it fails with output:
````
[2020-08-25][12:07:21][startup][INFO] [Row { columns: [Column { name: "id", column_type: Int4 }, Column { name: "value", column_type: Intn }], data: TokenRow { data: [I32(Some(1)), I32(Some(1))] } }, Row { columns: [Column { name: "id", column_type: Int4 }, Column { name: "value", column_type: Intn }], data: TokenRow { data: [I32(Some(2)), U8(None)] } 
}]
[2020-08-25][12:07:21][startup][INFO] a = Some(1)
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Conversion("cannot interpret U8(None) as an i32 value")'
````
So in some reason the value in a second row is parsed as U8(None), not i32(None). 
Is it a bug? If not, could you suggest some workaround?

**Comment by SpeedSX** [comment]
Investigated that a little bit, I found that happens due to a "feature" of a protocol. Sometimes Intn values are received as int of zero length, and decoder stores it as U8(None) (as it is not aware of actual field type). I'm not sure how to pass metadata to stream decoder. So far I just created a fork and changed implementation of FromSql for i32 adding to match ColumnData::U8(None) => Ok(None).
This should be done for all numeric type which can face this problem (i16, i64, ...?)
I don't think this solution is graceful enough to recommend so I'm leaving this issue open.
